src/datadog/datadog_agent_config.cpp

```diff
@@ -114,6 +114,22 @@ Expected<FinalizedDatadogAgentConfig> finalize_config(
   result.flush_interval =
       std::chrono::milliseconds(config.flush_interval_milliseconds);
 
+  if (config.request_timeout_milliseconds <= 0) {
+    return Error{Error::DATADOG_AGENT_INVALID_FLUSH_INTERVAL,
```

Comment:
The error code is not the right one because I would like to get your thought. How about a less specific error code? Like `DATADOG_AGENT_INVALID_CONFIGURATION`?

Reply:
I like having specific error messages. In most cases, when I'm writing new code and encounter an error condition, I add a new value to `enum class Error::Code`.